From d40170c9444cdb1598a641edc940eeccae0139b0 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 30 Oct 2016 14:57:21 +0000
Subject: [PATCH] applets/budgie-menu: Stop Vala unreffing the Settings object

This pertains to issue #632, and under strange conditions will result in
the settings object being unreffed one too many times, leading to use of
a dangling pointer, i.e. insta segfault

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/panel/applets/budgie-menu/BudgieMenuWindow.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/panel/applets/budgie-menu/BudgieMenuWindow.vala b/src/panel/applets/budgie-menu/BudgieMenuWindow.vala
index ed488e0..65cba2e 100644
--- a/src/panel/applets/budgie-menu/BudgieMenuWindow.vala
+++ b/src/panel/applets/budgie-menu/BudgieMenuWindow.vala
@@ -105,7 +105,7 @@ public class BudgieMenuWindow : Gtk.Popover
 
     protected int icon_size = 24;
 
-    public unowned Settings settings { public get; public set; }
+    public Settings settings { public get; public set; }
 
     private bool reloading = false;
 
-- 
2.7.4

